### ISSUE TITLE
Enable Preview Deployment Apps for easy review

### ISSUE THREAD

**Issue by nickvisut** [sfbrigade/datasci-earthquake#105]
## Context
Pain point: Team members can't currently see a Pull Request's version of an app without fiddling w/code (pulling down the branch and running it). This means there is friction when it comes to reviewing and manually testing, especially for code reviewers.

[Preview Apps](https://dev.to/livecycle/how-to-make-vercel-preview-deployments-10x-more-valuable-for-your-team-3m50)* for GitHub solve this. Integrating them **_automatically_** adds a link right in the PR that leads to a deployed app (pictured below). This would allow the entire team (not just devs) to take a look at a Pull Request's changes even before merging into the (sacred) default branch. This helps w/the feedback loop, code quality, and can help build team confidence for releases.
  _*encouraged reading_

<img width="922" alt="Screenshot 2024-11-24 at 6 26 57 PM" src="https://github.com/user-attachments/assets/a055f7de-34ff-4477-b07d-d07c79e4cc84">

Furthermore, it would lessen the need to send screenshots around and would take us closer to production deployments.

TLDR: Preview Apps allow the team to get feedback early and often and catch issues more easily before merge (without having to build the code locally, which is tedious and shuts out non-technical team members).

## Definition of Done
- [x] when a PR is submitted, automated checks are run prior to deploying Preview App
- [x] if checks pass, Preview App is deployed to Vercel
- [x] Preview App should show up as a link in the Pull Request for anyone to access
- [x] App behaves properly including all integrations (e.g., Mapbox) due to env variables being set correctly
- [x] Vercel account uses shared credentials

## Engineering Details
- [x] This will need to go into CI
- [x] If we use Vercel, we can start here [w/their docs](https://vercel.com/docs/deployments/preview-deployments). @mackcooper1408 mentions it's pretty straightforward to set up.

**Comment by nickvisut** [comment]
cc @mackcooper1408

**Comment by mackcooper1408** [comment]
this looks great, thanks for all the detail. Let's chat about getting a Vercel account setup with the shared email.

**Comment by nickvisut** [comment]
cc @agennadi bc CI-related

**Comment by nickvisut** [comment]
cc @mantuok bc also front end

**Comment by mantuok** [comment]
@nickvisut @mackcooper1408  Am I correct in understanding that we need to deploy to Vercel first to enable this feature? If so, should we create a separate issue for deployment and mark this one as blocked by it?

**Comment by nickvisut** [comment]
@mantuok that makes sense altho perhaps can also be created as a sub-issue

**Comment by nickvisut** [comment]
Movement to Vercel team account and the app being deployed in all its glory is due in large part to @agennadi.

I think one last thing is to test that our new API integration code works properly when deployed (once it is ready and merged in).

**Comment by nickvisut** [comment]
Added you as assignee @agennadi since you have been working the most on this and moving it to the finish line.

**Comment by nickvisut** [comment]
Noting here that it appears users need to log in to see previews. Need to look into whether this can be turned off. Public previews are preferred as that means anyone is able to help review a PR's deployed site.

**Comment by agennadi** [comment]
@nickvisut I enabled public previews

**Comment by nickvisut** [comment]
> [@nickvisut](https://github.com/nickvisut) I enabled public previews

Perfect, thank you! Closing this as completed.
